src/Utils.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import MetaProperty from "./MetaProperty";
+import { PathUtils } from "./PathUtils";
+import Constants from "./Constants";
+
+export class Utils {
+
+    /**
+     * Returns true if the editor is in Edit Mode
+     *
+     * @returns {Boolean} the result of the check of the Editor mode
+     */
+    public static isEditMode(): boolean {
+        const aemMode = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE);
+        if (aemMode != null) {
+            return aemMode === Constants.AEM_MODE_EDIT;
+        }
+        else {
+            if (PathUtils.isBrowser()) {
+                const param = Utils.getEditParam();
+                return param === Constants.AEM_MODE_EDIT;
+            }
+            else {
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Returns query parameter value for aem mode.
+     *
+     * @returns {String | null} the result of the check of the Editor mode
+     */
+    public static getEditParam() : string | null {
+        return new URL(document.location.href).searchParams.get(Constants.AEM_MODE_KEY);
+    }
+
+    /**
+     * Appends javascript to the page
+     */
+    public static appendScripts(clientLibUrl : string) : void {
```

Comment:
`appendScripts` and `appendStyleSheets` could be mutualized into a private `appendTag` function that requires the name of the element and tag name that will be created

Reply:
`appendTag(tagName, attrs)` - as script expects `src`, while link `href`, plus `attrs` would give us further flexibility for the future, wdyt?

Reply:
Do we have a lot of usecases where we need to add various tags? To me it looks like a little bit of over-optimization considering the usecase it caters. But I don't have a stong opinion against creating a function like  `appendTag(tagName, attrs)`.